### PR TITLE
TP178504 | Added notification service extension | Updated readme with rich push

### PR DESCRIPTION
**Pull request by ajaysubra**
# Description

This PR does two things - 

1. Update the readme with rich push notification instructions.
2. Update the example app with sample code that customers can directly add to their app to get rich push setup.

Pending - 

1. Need to test cocoa pods integration.

# Check List

- [ ] Are you changing anything with the public API?
- [ ] Have you tested this change on real device?
- [ ] Are your changes backwards compatible with previous SDK Versions?
- [ ] Have you added unit test coverage for your changes?
- [ ] Have you verified that your changes are compitable with all the operating system version this SDK currently supports?

# Manual Test Plan

1.

# Supporting Materials

### CODE REVIEW

**Comment by klaviyoit** [comment]
https://klaviyo.tpondemand.com/entity/178504

File: Examples/KlaviyoSwiftExamples/SPMExample/NotificationServiceExtension/NotificationService.swift [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            /// Create a file URL based on the image identifier.
+            let fileURL = directoryPath.appendingPathComponent(identifier)
+
+            /// Create a Data object from the image.
+            guard let imageData = image.pngData() else {
+                return nil
+            }
+
+            /// Attempt to write the file to disk.
+            try imageData.write(to: fileURL)
```
Comment (evan-masseau) [inline Examples/KlaviyoSwiftExamples/SPMExample/NotificationServiceExtension/NotificationService.swift:132]: I feel like you mentioned this somewhere, are we limited in storage space, should we remove old images somehow?

File: Examples/KlaviyoSwiftExamples/SPMExample/NotificationServiceExtension/NotificationService.swift [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+/// When push payload is marked as there being mutable-content this service
+/// (more specifically the `didReceiveNotificationRequest` ) is called to perform
+/// tasks such as downloading images and attaching it to the notification before it's displayed to the user.
+///
+/// There is a limited time before which `didReceiveNotificationRequest`  needs to wrap up it's operations
+/// else the notification is displayed as received.
+///
+/// Any property from `UNMutableNotificationContent` can be mutated here before presenting the notification.
+class NotificationService: UNNotificationServiceExtension {
```
Comment (evan-masseau) [inline Examples/KlaviyoSwiftExamples/SPMExample/NotificationServiceExtension/NotificationService.swift:21]: Just curious does this code change much for gif/video? Or will this work once the server supports sending those?

File: README.md [hunk 3]
```diff
@@ -332,6 +332,62 @@ Additionally, you can also locally trigger a deep link to make sure your code is
...
+
+### Step 2: Implement the notification service app extension
+
+The notification service app extension is responsible for downloading the media resource and attaching it to the push notification.
+
+Once step 1 is complete, you should see a file called `NotificationService.swift` under the notification service extension target. You can either implement it using Apple's documentation [here](https://developer.apple.com/documentation/usernotifications/modifying_content_in_newly_delivered_notifications) or use our sample code from [here](Examples/KlaviyoSwiftExamples/SPMExample/NotificationServiceExtension/NotificationService.swift).
+
+### Step 3: Test your rich push notifications
+
+#### Local testing
```
Comment (evan-masseau) [inline README.md:360]: Can it be tested from a terminal?

**Review by evan-masseau — APPROVED** [review]
looks good!

File: Examples/KlaviyoSwiftExamples/SPMExample/NotificationServiceExtension/NotificationService.swift (hunk 2 above)
Comment (ajaysubra) [reply Examples/KlaviyoSwiftExamples/SPMExample/NotificationServiceExtension/NotificationService.swift:21]: Good call. I just updated the code to work in  a media agnostic way. Passing down the media type so we don't hardcode it to png but instead use what app sends and if it doesn't send any we default to png.

File: README.md (hunk 3 above)
Comment (ajaysubra) [reply README.md:360]: Yeah I tested this on the terminal using `xcrun simctl push booted com.klaviyo.spm.example.SPMExample ~/Desktop/payload.json`  didn't work too well for me. 


Here's the payload file - 

```
{
  "aps": {
    "alert": {
      "title": "Free apple vision pro" ,
      "body": "Free Apple vision pro when you buy a Klaviyo subscription"
    },
    "mutable-content": 1
  },
  "rich-media": "https://www.apple.com/v/apple-vision-pro/a/images/overview/hero/portrait_base__bwsgtdddcl7m_large.jpg",
"rich-media-type":"jpg",
  "Simulator Target Bundle" : "com.klaviyo.spm.example.SPMExample"
}
```

Please let me know if it works for you. I might be missing something.

File: Examples/KlaviyoSwiftExamples/SPMExample/NotificationServiceExtension/NotificationService.swift (hunk 1 above)
Comment (ajaysubra) [reply Examples/KlaviyoSwiftExamples/SPMExample/NotificationServiceExtension/NotificationService.swift]: Yeah. I did try and look this up but the documentation from apple on this is very vague and not much on SO either. I'm going to create a task for us to look into this but for now I'm going to go ahead with merging this code.

**Pull request merged**